### PR TITLE
Add more info to TwoFactorChallengeFailedException

### PR DESCRIPTION
**Pull request by Haacked**
I was running into an issue where i wanted more information from the
TwoFactorChallengeFailedException. It turns out, the exception could
easily provide both the TwoFactorType AND the authentication code
provided. So :boom:!

### CODE REVIEW

File: Octokit/Exceptions/TwoFactorChallengeFailedException.cs [hunk 1]
```diff
@@ -37,9 +39,16 @@ public override string Message
...
+
+        static TwoFactorType ParseTwoFactorType(ApiException exception)
+        {
+            return exception == null ? TwoFactorType.None : Connection.ParseTwoFactorType(exception.HttpResponse);
+        }
+
 #if !NETFX_CORE
         /// <summary>
-        /// Constructs an instance of TwoFactorChallengeFailedException
+        /// Constructs an instance of TwoFactorRequiredException.
```
Comment (shiftkey) [inline Octokit/Exceptions/TwoFactorChallengeFailedException.cs]: :memo: this should remain as `TwoFactorChallengeFailedException`

File: Octokit/Exceptions/TwoFactorAuthorizationException.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+            TwoFactorType = twoFactorType;
+        }
+
+        /// <summary>
+        /// Expected 2FA response type
+        /// </summary>
+        public TwoFactorType TwoFactorType { get; private set; }
+
+        #if !NETFX_CORE
```
Comment (shiftkey) [inline Octokit/Exceptions/TwoFactorAuthorizationException.cs]: :lipstick: indenting

File: Octokit/Exceptions/TwoFactorChallengeFailedException.cs (hunk 1 above)
Comment (Haacked) [inline Octokit/Exceptions/TwoFactorChallengeFailedException.cs]: Nice catch!

**Comment by Haacked** [comment]
:spaghetti:

**Pull request merged**